tests/e2e/test_operator.py

```diff
@@ -2907,6 +2908,80 @@ def test_035(self):
             self.context.shell.close()
 
 
+@TestScenario
+@Requirements(RQ_SRS_026_ClickHouseOperator_Managing_ReprovisioningVolume("1.0"))
+@Name("test_036. Check operator volume re-provisioning")
+def test_036(self, num_retries=100):
+    """Check clickhouse operator recreates volumes and schema if volume is broken."""
+    cluster = "simple"
+    manifest = f"manifests/chi/test-036-volume-re-provisioning.yaml"
+    chi = yaml_manifest.get_chi_name(util.get_full_path(manifest))
+    util.require_keeper(keeper_type=self.context.keeper_type)
+
+    with Given("I get terminal shell"):
+        self.context.shell = get_shell()
+
+    with And("chi exists"):
+        kubectl.create_and_check(
+            manifest=manifest,
+            check={
+                "apply_templates": {settings.clickhouse_template},
+                "pod_count": 1,
+                "do_not_delete": 1,
+            },
+        )
+
+    with And("I create replicated table with some data"):
+        create_table = """
+            CREATE TABLE test_local_036 ON CLUSTER 'simple' (a UInt32)
+            Engine = ReplicatedMergeTree('/clickhouse/{installation}/tables/{shard}/{database}/{table}', '{replica}')
+            PARTITION BY tuple()
+            ORDER BY a
+            """.replace('\r', '').replace('\n', '')
+        clickhouse.query(chi, create_table)
+        clickhouse.query(chi, f"INSERT INTO test_local_036 select * from numbers(10000)")
+
+    with When("I delete PV", description="delete PV on replica 1"):
+        pv_name = self.context.shell("kubectl get pv | grep test/disk1-chi-test-036-volume-re-provisioning-simple-0-1-0").output.split()[0]
+
+        self.context.shell(f"kubectl delete pv {pv_name} --force &")
+        self.context.shell("""kubectl patch pv $(kubectl get pv | grep test/disk1-chi-test-036-volume-re-provisioning-simple-0-1-0 
+            | awk '{print $1}') -p '{"metadata":{"finalizers":null}}' """.replace('\r', '').replace('\n', ''))
+
+    with And("I check replica is deleted and recreated"):
```

Comment:
I think the check for ZooKeeper contents is not reliable in general. It is sufficient to test that PV is created back and data is properly replicated. It can not be done without handling replica properly in ZooKeeper anyway.